src/libraries/Directory.Build.targets

```diff
@@ -233,4 +233,22 @@
     <None Include="$(MSBuildProjectDirectory)\**\*.cs"
           Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Compile)" />
   </ItemGroup>
+
+  <!--Instructs compiler not to emit .locals init, using SkipLocalsInitAttribute.-->
+  <Choose>
+    <When Condition="'$(ShouldSkipLocalsInit)' == 'true' and '$(IsNETCoreApp)' == 'true'">
+      <PropertyGroup >
+        <!-- This is needed to use the SkipLocalsInitAttribute. -->
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <Compile Include="$(CommonPath)ShouldSkipLocalsInit.cs" Link="Common\ShouldSkipLocalsInit.cs" />
+      </ItemGroup>
+      <ItemGroup Condition="$(TargetFramework.StartsWith('netstandard')) or $(TargetFramework.StartsWith('net4')) or '$(TargetFramework)' == 'netcoreapp3.0' or '$(TargetFrameworkIdentifier)' == '.NETFramework'">
+        <Compile Include="$(CoreLibSharedDir)System\Runtime\CompilerServices\SkipLocalsInitAttribute.cs" Link="System\Runtime\CompilerServices\SkipLocalsInitAttribute.cs" />
```

Comment:
We weren’t ILLink’ing these projects before, right? So should we be skipping init locals on them now?

Reply:
While it is technically possible we were doing this (meaning we could have had a library part of the shared framework that didn't target netcoreapp) I am also curious why you had to add this. Can we check a netfx build log to see if we were running IlLink there?

Reply:
Yup, will adjust the conditions to only skip init locals on projects that we were included previously.